dot/build_spec_integration_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+//go:build integration
+// +build integration
+
+package dot
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildFromGenesis(t *testing.T) {
+	t.Parallel()
+
+	file, err := genesis.CreateTestGenesisJSONFile(false)
+	defer os.Remove(file)
+	require.NoError(t, err)
+	bs, err := BuildFromGenesis(file, 0)
+
+	const expectedChainType = "TESTCHAINTYPE"
+	expectedProperties := map[string]interface{}{
+		"ss58Format":    0.0,
+		"tokenDecimals": 0.0,
+		"tokenSymbol":   "TEST",
+	}
+
+	bs.genesis.ChainType = expectedChainType
+	bs.genesis.Properties = expectedProperties
+
+	require.NoError(t, err)
+
+	// confirm human-readable fields
+	hr, err := bs.ToJSON()
+	require.NoError(t, err)
+	jGen := genesis.Genesis{}
+	err = json.Unmarshal(hr, &jGen)
+	require.NoError(t, err)
```

Comment:
Why are we unmarshaling then marshaling?

Reply:
This is taking the build spec `bs` and converting to JSON, then using that JSON to build the Genesis object.  Also, this PR just moves xx_test files to xx_integration_test files, so this code was not changed.

Reply:
Shouldn't this JSON marshaling be taken care of in unit tests instead?